src/main/java/org/skriptlang/skript/lang/converter/ChainedConverter.java

```diff
@@ -0,0 +1,65 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.lang.converter;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * A Chained Converter is very similar to a regular {@link Converter}.
+ * They are used when it is not possible to directly convert from Type A to Type B.
+ * Instead, a "middle man" is used that makes it possible to convert from Type A to Type B.
+ * What this means is that Type A is converted to the "middle man" type.
+ * Then, that "middle man" type is converted to Type B.
+ *
+ * Of course, multiple Chained Converters can be chained together to provide far more conversion possibilities.
+ * In these cases, you might have, for example, another "middle man" between
+ *  this Chained Converter's Type A and "middle man"
+ *
+ * By using conversion chains, Skript can convert an object into a different type that seemingly has no relation.
+ *
+ * @param <From> The type to convert from.
+ * @param <Middle> A middle type that is needed for converting 'from' to 'to'.
+ *                'from' will be converted to this type, and then this type will be converted to 'to'.
+ * @param <To> The type to convert to.
+ */
+final class ChainedConverter<From, Middle, To> implements Converter<From, To> {
```

Comment:
Generic full names are rarely used. Mainly because it's misleading as to if it's an actual class object or not.

Reply:
We discussed this and felt that the reasoning from the original oracle style guide (where this tradition started) was outdated from when there were no IDE insights.
If you refer to style guides like google's they have also abandoned single-letter names.
Walrus said there was no rule in Skript's code style so we have no landmark on this.

Reply:
Google's Java Style Guide states nothing about it being abandoned nor that it should follow naming. The guide states to use single letters, and if you must use naming, ensure you have a generic type letter after it. 
![Screenshot_20220807-113741](https://user-images.githubusercontent.com/16087552/183303915-7ba5ebc5-6f31-47d2-8502-aed968124e18.png)
We should be using single letter generics.

Reply:
I think the usage of more descriptive type variable names is beneficial for these systems. Something we can discuss as the whole team though :)

Reply:
IMO in some usage it may be better to use full names so this is okay for me